backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -226,7 +225,11 @@ public boolean updateEventTypeBehaviors(String accountId, UUID eventTypeId, Set<
                  */
                 String insertQuery = "INSERT INTO event_type_behavior (event_type_id, behavior_group_id, created) " +
                         "SELECT :eventTypeId, :behaviorGroupId, :created " +
-                        "WHERE EXISTS (SELECT 1 FROM behavior_group WHERE account_id = :accountId AND id = :behaviorGroupId) " +
+                        "WHERE EXISTS (" +
+                            "SELECT 1 FROM behavior_group bg, applications a, event_type et " +
+                            "WHERE bg.account_id = :accountId AND bg.id = :behaviorGroupId " +
+                            "AND bg.bundle_id = a.bundle_id AND a.id = et.application_id AND et.id = :eventTypeId" +
```

Comment:
This is a short-term fix for the issue raised on Slack but I think we should revisit this fix later. Ideally, the endpoint should return a 400 with a message containing the exact cause (i.e. the behaviorGroupIds that weren't accepted here).

However, the validation has to be smart enough to avoid putting too much pressure on the DB. If possible, we should avoid iterating over the behaviorGroupIds list and run a SQL query to validate each one of them.

Reply:
Could we just do a query like (outside the loop) `SELECT COUNT(*) FROM behavior_group bg WHERE bg.id IN (:behaviorGroupId)` and throw a bad request if the count is different than the input.

Additionaly, if we want to provide a more detailed error we could use a similar query fetching the behavior group.